src/api/Metadata.js

```diff
@@ -826,12 +827,19 @@ class Metadata extends File {
                 const cache: APICache = this.getCache();
                 const key = this.getMetadataCacheKey(id);
                 const metadata = cache.get(key);
-                metadata.editors.splice(
-                    metadata.editors.findIndex(
-                        editor => editor.template.scope === scope && editor.template.templateKey === templateKey,
-                    ),
-                    1,
-                );
+                if (isMetadataRedesign) {
+                    metadata.templateInstances.splice(
+                        metadata.templateInstances.findIndex(instance => instance.id === template.id),
```

Comment:
Shouldn't compare `instance.templateId` and `template.id`? See my comment [here](https://github.com/box/box-ui-elements/pull/3646#discussion_r1761052708)

Reply:
Alternatively, we could just pass `tempateInstance.id` and then look for it's index to be more precise

Reply:
I also wonder if we need to check scope. Do you know why it's needed when finding index of instance in editor?

Reply:
OK, so I see later in the code that you actually pass `MetadataInstance` as `template`. This may be confusing and the types do not match. Can you fix types and maybe add some comment explaining what's going on here?